### PR TITLE
MM-32347 Site Subsections

### PR DESCRIPTION
**Pull request by anurag6713**
#### Summary
Site Configuration
* Customization
* Localization
* Users and Teams
* Notifications
* Announcement Banner
* Emoji
* Posts
* File Sharing and Downloads
* Public Links
* Notices

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32347

### Screenshot
Screenshot 2021-03-09 at 19 58 24

### CODE REVIEW

**Comment by sbishel** [comment]
Looks good, but a failing test.

**Review by sbishel — APPROVED** [review]

File: components/admin_console/admin_definition.jsx [hunk 1]
```diff
@@ -188,7 +188,9 @@ export const it = {
         }
         return cloud.subscription.is_paid_tier === 'true';
     },
-    userHasReadPermissionOnResource: (key) => (config, state, license, enterpriseReady, consoleAccess) => consoleAccess?.read?.[key],
+    userHasReadPermissionOnResource: (key) => (config, state, license, enterpriseReady, consoleAccess) => {
+        return consoleAccess?.read?.[key];
+    },
```
Comment (hahmadia) [inline components/admin_console/admin_definition.jsx:193]: I believe this change can be reverted. I think keeping it as a one liner fits better with the rest.

File: e2e/cypress/support/api/role.js [hunk 2]
```diff
@@ -198,12 +216,30 @@ export const defaultRolesPermissions = {
...
+        'sysconsole_read_site_customization',
+        'sysconsole_read_site_localization',
+        'sysconsole_read_site_users_and_teams',
+        'sysconsole_read_site_notifications',
+        'sysconsole_read_site_announcement_banner',
+        'sysconsole_read_site_emoji',
+        'sysconsole_read_site_posts',
+        'sysconsole_read_site_file_sharing_and_downloads',
+        'sysconsole_read_site_public_links',
+        'sysconsole_read_site_notices',
```
Comment (hahmadia) [inline e2e/cypress/support/api/role.js:242]: Hey Anurag,

Please don't forget to give any ancillary permissions that are associated to these! Thanks.

File: components/admin_console/admin_definition.jsx (hunk 1 above)
Comment (anurag6713) [reply components/admin_console/admin_definition.jsx:193]: Have no idea how this happened 🧙 ... Reverting now.

File: e2e/cypress/support/api/role.js [hunk 3]
```diff
@@ -93,7 +102,16 @@ export const defaultRolesPermissions = {
...
+        'sysconsole_write_site_customization',
+        'sysconsole_write_site_localization',
+        'sysconsole_write_site_users_and_teams',
+        'sysconsole_write_site_notifications',
+        'sysconsole_write_site_announcement_banner',
+        'sysconsole_write_site_emoji',
+        'sysconsole_write_site_posts',
+        'sysconsole_write_site_file_sharing_and_downloads',
+        'sysconsole_write_site_public_links',
+        'sysconsole_write_site_notices',
```
[the remark this answers is not in the document]
Comment (anurag6713) [reply e2e/cypress/support/api/role.js:113]: My ancillary permission `edit_brand` is already mentioned in list.

**Review by sbishel — APPROVED** [review]

**Review by hahmadia — APPROVED** [review]

**Comment by sbishel** [comment]
Ran into an issue during testing. When. clicking on `Users and Teams` A javascript error is thrown due to the fact that `FeatureFlags` are not being returned in the config.

**Comment by furqanmlk** [comment]
@sbishel ```Users and Teams``` page is still empty 

image

**Comment by sbishel** [comment]
> @sbishel `Users and Teams` page is still empty
> 
> image

This is due to the issue stated above. "FeatureFlags" are not being returned in the Admin config.
Created a separate issue for this. 
https://mattermost.atlassian.net/browse/MM-33905

**Review by furqanmlk — APPROVED** [review]

**Pull request merged**